lib/src/rules/unnecessary_overrides.dart

```diff
@@ -106,14 +120,27 @@ abstract class _AbstractUnnecessaryOverrideVisitor extends SimpleAstVisitor {
   visitSuperExpression(SuperExpression node) {
     rule.reportLint(declaration.name);
   }
+
+  bool _haveSameDeclaration() {
+    if (declaration.element.returnType != inheritedMethod.returnType)
+      return false;
+    for (var i = 0; i < inheritedMethod.parameters.length; i++) {
+      final superParam = inheritedMethod.parameters[i];
+      final param = declaration.element.parameters[i];
+      if (param.type != superParam.type) return false;
+      if (param.name != superParam.name) return false;
+      if (param.isCovariant != superParam.isCovariant) return false;
```

Comment:
We should check the optionality of the parameters. Should we also check the default value for option parameters?